src/main/java/dev/slimevr/vr/trackers/TrackerPosition.java

```diff
@@ -1,87 +1,127 @@
 package dev.slimevr.vr.trackers;
 
+import java.lang.reflect.Modifier;
 import java.util.EnumMap;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Optional;
+import java.util.logging.Level;
 
+import solarxr_protocol.datatypes.BodyPart;
 
-public enum TrackerPosition {
 
-	NONE(0, "", TrackerRole.NONE),
-	HMD(1, "HMD", TrackerRole.HMD),
-	NECK(2, "body:neck", null),
-	CHEST(3, "body:chest", TrackerRole.CHEST),
-	WAIST(4, "body:waist", TrackerRole.WAIST),
-	HIP(5, "body:hip", null),
-	LEFT_KNEE(6, "body:left_knee", TrackerRole.LEFT_KNEE),
-	RIGHT_KNEE(7, "body:right_knee", TrackerRole.RIGHT_KNEE),
-	LEFT_ANKLE(8, "body:left_ankle", null),
-	RIGHT_ANKLE(9, "body:right_ankle", null),
-	LEFT_FOOT(10, "body:left_foot", TrackerRole.LEFT_FOOT),
-	RIGHT_FOOT(11, "body:right_foot", TrackerRole.RIGHT_FOOT),
-	LEFT_CONTROLLER(12, "body:left_controller", TrackerRole.LEFT_CONTROLLER),
-	RIGHT_CONTROLLER(13, "body:right_controller", TrackerRole.RIGHT_CONTROLLER),
-	LEFT_FOREARM(14, "body:left_forearm", TrackerRole.LEFT_ELBOW),
-	RIGHT_FOREARM(15, "body:right_forearm", TrackerRole.RIGHT_ELBOW),
-	LEFT_UPPER_ARM(16, "body:left_upperarm", null),
-	RIGHT_UPPER_ARM(17, "body:right_upperarm", null),
-	LEFT_HAND(18, "body:left_hand", TrackerRole.LEFT_HAND),
-	RIGHT_HAND(19, "body:right_hand", TrackerRole.RIGHT_HAND),;
+/**
+ * Represents a position on the body that a tracker could be placed. Any bone is
+ * a valid position.
+ *
+ * TrackerPosition intentionally lacks a numerical id to avoid breakage.
+ */
+public enum TrackerPosition {
+	// @formatter:off
+	HMD("HMD", TrackerRole.HMD, BodyPart.HMD),
+	NECK("body:neck", TrackerRole.NECK, BodyPart.NECK),
+	CHEST("body:chest", TrackerRole.CHEST, BodyPart.CHEST),
+	WAIST("body:waist", Optional.empty(), BodyPart.WAIST),
+	HIP("body:hip", TrackerRole.WAIST, BodyPart.HIP),
```

Comment:
Someone please confirm that this reassignment makes sense:

* reassigned `TrackerPosition.HIP` to `TrackerRole.WAIST`
* reassigned `TrackerPosition.WAIST` to `Optional.empty()`

I did this because I thought the slime waist tracker should not be treated as the steamvr hip tracker

Reply:
that works, but I don’t even know if this (linking steamvr tracker role to slimevr tracker position) really has a use tbh?
Maybe Eiren can answer.

Anyway, this is something I’ve been looking at and kinda wanting to change, so I approve :p

Reply:
Why would the waist tracker be linked to a steamvr tracker? We would use hip for steamvr waist, and chest for steamvr chest

Reply:
Assuming a user has all possible SlimeVR trackers, yes, the hip would influence the SteamVR waist tracker directly.

The thing is that I fail to even see why we are linking the SteamVR trackers and SlimeVR ones directly in the enums like that… **Maybe I’m missing something** but this doesn’t seem like a good thing to do? If user has waist but no hip, we could try to get hip and not find anything. Same with the other way around btw!

Maybe this is something you could look into in another PR?

Reply:
But if user has waist but no hip, it still moves the hip bone. So when determining the vive tracker position in steamvr, we would still use the hip. It's just that the hip is being rotated/translated because of it's parent bones, the waist